frontend/ui/data/keyboardlayouts/ja_keyboard.lua

```diff
@@ -28,11 +28,11 @@ local DEFAULT_KEITAI_TAP_INTERVAL = 2
 -- information.
 
 local function getKeitaiTapInterval()
-    return G_reader_settings:readSetting("keyboard_japanese_keitai_tap_interval") or DEFAULT_KEITAI_TAP_INTERVAL
+    return time.s(G_reader_settings:readSetting("keyboard_japanese_keitai_tap_interval", DEFAULT_KEITAI_TAP_INTERVAL_S))
 end
 
 local function setKeitaiTapInterval(interval)
-    G_reader_settings:saveSetting("keyboard_japanese_keitai_tap_interval", interval)
+    G_reader_settings:saveSetting("keyboard_japanese_keitai_tap_interval", time.toS(interval))
```

Comment:
`time.s()` looks nice, but the change of case for `s` in `time.toS()` is a bit bothering (from a purely aestheticanalretentiveness point of view :)
Dunno if these alternatives could be better ?
```lua
time.S()   time.toS()
time.s()   time.to_s()  -- even if our coding style prevent underscore in favor of camelCase for function names
time.s()   time.2s()    -- looks like this won't be accepted by Lua
```

Reply:
I like the `snake_case` version myself ;).

Plus, doesn't the style guide say that of `methods`? This is technically a `function`, as there isn't really a custom data object anymore, just a table of functions ;).

Reply:
+1 for the snake_case.

Reply:
What shall we do with `getDurationMs` and friends?
`get_duration_ms`?

What's with `time.tonumber` as it seems a bit odd to have a lot of `time.to_xxx`, but `time.tonumber` and `time.fromnumber`?
Shall we rename `time.fromnumber` to `time.number` (and maybe add `time.fromnumber = time.number`) and rename function `time.tonumber` in `time.to_number` (and for consistency with the other conversion methods `time.tonumber = time.to_number`)?

Reply:
Hmm, that's a good question.

FWIW, the `tonumber` thing is probably to mimic Lua's native `tostring` function ;).

The duration stuff is probably easier to deal with, I'd say go with snake_case there, too (if only to make sure units are always written the same way across the module singatures).

Reply:
And adding `time.to_number = time.tonumber` would leave us both options open.

Reply:
If we go with `time.to_s()` (I'm ok with that), let's have all `to_` with the underscore (which looks like an arrow `to->` :), so `to_number()`.
And then, `time.number(123)`.
But what's the expected number ? What does it mean, without any time unit ? Or is this equivalent to `time.s(123)` ?

I would also avoid "aliases" and giving multiple names to the same functions - to keep our code consistent.
`get_duration_ms(some_previous_time)` : can't it become `time.to_ms(time.duration(some_previous_time))` ? which feels a bit long and tedious.
`time.duration_to_ms(some_previous_time)` then, as a shortcut ?
or `time.duration_as_ms(some_previous_time)`, if this reads as better english?

`time.duration(some_previous_time)` would then give a time object (an obscure fts actually, if I get all this right :) I'll have to learn it all once it's done)

Reply:
`duration_in_<unit>` would make sense, then, yeah ;).

Reply:
> If we go with `time.to_s()` (I'm ok with that), let's have all `to_` with the underscore (which looks like an arrow `to->` :), so `to_number()`. 

:+1:

>And then, `time.number(123)`. But what's the expected number ? What does it mean, without any time unit ? Or is this equivalent to `time.s(123)` ?

You are absolutely right (without a unit it is nuisance), I will drop `time.number` in favor of `time.s`.

> I would also avoid "aliases" and giving multiple names to the same functions - to keep our code consistent. 

:+1:  So no `time.number` as it does the same as `time.s`

> `get_duration_ms(some_previous_time)` : can't it become `time.to_ms(time.duration(some_previous_time))` ? which feels a bit long and tedious. `time.duration_to_ms(some_previous_time)` then, as a shortcut ? or `time.duration_as_ms(some_previous_time)`, if this reads as better english?

We have `get_duration_ms` 5 times in the code and `get_duration_us` one time. So I will replace t `duration_xs` with `time.to_xs(get_duration(...))`.


> `time.duration(some_previous_time)` would then give a time object (an obscure fts actually, if I get all this right :) I'll have to learn it all once it's done)

Yes. An the usage is really straight forward (compared to TimeVal).
I will append some usage infos at the docs part of `time.lua`.

Reply:
Can't we have / do we really not want to have methods to these time objects, even if there are plain Lua numbers ?
Would metatable stuff allow it ? Would it add too much overhead ?
```lua
last_action_time = time.now()
[ do some stuff]
time.now().duration_since(last_action_time).to_ms()
```
would read quite nice :)

Reply:
Would require metatables, yeah, which is a non-goal of this ;).

(And would read something like `time.now:duration_since(then):to_ms()`; because methods ;p).

Sidebar: I've never been of fan of that sort( of syntax anyway ^^.

Reply:
IMHO: For low level things, and things that should not produce overhead, functions are more clear to read (you see what is happening).
For complicated objects, methods are more readable.
(a bit simplified)